lib/go/internal/v0/edgecontext.go

```diff
@@ -0,0 +1,259 @@
+package v0
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"sync/atomic"
+	"time"
+
+	"github.com/apache/thrift/lib/go/thrift"
+	"github.com/reddit/baseplate.go/detach"
+	"github.com/reddit/baseplate.go/ecinterface"
+	"github.com/reddit/baseplate.go/log"
+	"github.com/reddit/baseplate.go/secrets"
+	"github.com/reddit/baseplate.go/timebp"
+	"github.com/reddit/edgecontext/lib/go/ecdata"
+
+	ecthrift "github.com/reddit/edgecontext/lib/go/internal/reddit/edgecontext"
+)
+
+func init() {
+	copyEC := func(dst, src context.Context) context.Context {
+		if ec, ok := GetEdgeContext(src); ok {
+			dst = SetEdgeContext(dst, ec)
+		}
+		return dst
+	}
+
+	detach.Register(detach.Hooks{
```

Comment:
🔕 with `context.WithoutCancel`added in go 1.21 I don't think we are using `detach` any more. but we can clean this up in follow up PR instead.